docs/data/material/components/init-color-scheme-script/init-color-scheme-script.md

```diff
@@ -0,0 +1,140 @@
+---
+productId: material-ui
+title: InitColorSchemeScript component
+components: InitColorSchemeScript
+githubSource: packages/mui-material/src/InitColorSchemeScript
+---
+
+# InitColorSchemeScript
+
+<p class="description">InitColorSchemeScript component removes the dark mode flicker for server-side rendering application.</p>
+
+## Introduction
+
+`InitColorSchemeScript` component is used to remove the dark mode flicker for server-side rendering (SSR) applications.
+It is a script that runs before React to attach an attribute based on user preference.
+
+It is recommended to use this component whenever:
+
+- The application supports light and dark modes.
+- The application is server rendered (SSR).
+
+## Basics
+
+You need to enable CSS variables with `colorSchemeSelector: 'data'` in your theme.
+
+```js
+import { ThemeProvider, createTheme } from '@mui/material/styles';
+
+const theme = createTheme({
+  cssVariables: {
+    colorSchemeSelector: 'data',
+  },
+});
+
+function App() {
+  return <ThemeProvider theme={theme}>{/* Your app */}</ThemeProvider>;
+}
+```
+
+Then, render the `InitColorSchemeScript` component as the first child of the `<body>` tag.
+
+Below are the location for rendering `InitColorSchemeScript` component for a specific framework.
+
+### Next.js App Router
+
+Place the `InitColorSchemeScript` component in the `layout.tsx` file:
+
+```js title="src/app/layout.tsx"
+import InitColorSchemeScript from '@mui/material/InitColorSchemeScript';
+
+export default function RootLayout(props: { children: React.ReactNode }) {
+  return (
+    <html lang="en" suppressHydrationWarning>
+      <body>
+        <InitColorSchemeScript attribute="class" />
+        {props.children}
+      </body>
+    </html>
+  );
+}
+```
+
+### Next.js Pages Router
+
+Place the `InitColorSchemeScript` component in the `_document.tsx` file:
```

Comment:
nitpick: I think it's helpful to indicate that `_document` is a custom file that you have to create if you don't already have one.

```suggestion
Place the `InitColorSchemeScript` component in a custom `_document` file:
```